src/tests/sqlsmith/src/lib.rs

```diff
@@ -87,6 +87,11 @@ struct SqlGenerator<'a, R: Rng> {
     /// We might not read from all tables.
     bound_relations: Vec<Table>,
 
+    /// Relations in generated in 'From' statement shall be added to bound_relations and
+    /// bound_columns after all the relations are generated because they are not useable among
+    /// the relations
+    parallel_relations: Vec<Table>,
```

Comment:
I don't get it. Why do we need "parallel" relations? What is it "parallel" for?

Reply:
```
select * 
from t1, (select * from t2 where t1.x = t2.x);
```
This is invalid, because at the subquery, t1 is not allowed to be used yet but then add relation to context is called during every generator.
That why I called them "parallel", there may be better name for it, I can,t think of it.

Therefore, I solution here is to add them in a temporary buffer (which is parallel_relations), and only after all the relations in from is generated, only then I add them into the bounded col and bounded relations.

Reply:
I think Noel suggest lateral_contexts, i will use it isntead of parallel_relations

Reply:
Thinking about this more. I think this may add more complexity to context. It is harder to reason about now.

Maybe an alternative would be to:
1. Explicitly return tables from `gen_from_relation`:
    ```rs
    gen_from_relation(&mut self) -> (TableWithJoins, Table)
    ```
    This also means updating `gen_simple_table, gen_time_window_func` and other called to return `bound_relations`, instead of implicitly calling `add_relation_to_context`.

2. As a result only one place should call `add_relation_to_context`, which is `gen_from`.
    It will collate each `gen_from_relation`'s `bound_relations`, then add them to context.
    Currently there are 4 separate instances of `add_relation_to_context`. After this, there will be just one.